components/chart/header/index.tsx

```diff
@@ -88,12 +79,25 @@ const HeaderWapper = styled.div`
 export default function Header({
   showModulesSearch,
   chartName,
-  predefinedTags,
 }: {
   showModulesSearch: (state: boolean) => void;
   chartName: string;
-  predefinedTags: string[];
 }) {
+  useEffect(() => {
+    // Check if the URL has a 'token' query parameter
+    const urlParams = new URLSearchParams(window.location.search);
+    const token = urlParams.get('token');
+
+    if (token) {
+      storeToken(token);
+      // Remove the 'token' query parameter from the URL
+      const newUrl = `${window.location.origin}${window.location.pathname}`;
+      window.history.replaceState({}, document.title, newUrl);
+    }
+  });
```

Comment:
We should not use useEffect without dependencies,  and there is no need to add the code here, I prefer if you design a custom hook that does that and use it in index.ts inside TemplateRouter, you can call it maybe "useTokenChecker". And you can use https://nextjs.org/docs/pages/api-reference/functions/use-router to get and mutate the querystring

The useEffect inside the custom hook can use the router.query as depedency